rcgen/src/lib.rs

```diff
@@ -937,8 +937,9 @@ impl CertificateParams {
 			} else {
 				let hash = digest::digest(&digest::SHA256, pub_key.raw_bytes());
 				// RFC 5280 specifies at most 20 bytes for a serial number
-				let sl = &hash.as_ref()[0..20];
-				writer.next().write_bigint_bytes(sl, true);
+				let mut sl = hash.as_ref()[0..20].to_vec();
```

Comment:
We can obtain the same result (more or less) with alternatives:
* `let sl = &hash.as_ref()[..19]`. This will avoid one line and `mut`, but we are "wasting" 7 bits
* Avoid heap allocation with array and `memcpy`.

I opted for the most coincided approach with `Vec`, considering I don't expect  hot-path here